File: 진우/12weeks/13241. 최소공배수/최소공배수.py
# 유클리드 호제법
# 두 정수 a, b의 최대공약수를 반환한다.
def gcd(a, b):
    while b != 0:            # b가 0이 될 때까지 반복
        a, b = b, a % b      # a ← b, b ← a % b (유클리드 호제법 핵심)
    return a                 # b가 0이 되면 a가 최대공약수

A, B = map(int, input().split())             # 두 자연수 A, B 입력

g = gcd(A, B)                  # 두 수의 최대공약수 계산
result = (A * B) // g
print(result)

File: 진우/12weeks/13241. 최소공배수/최소공배수.cc
#include <iostream>

using namespace std;

long long Gcd(long long a, long long b) {
	while (b != 0) {
		long long temp = a;
		a = b;
		b = temp % b;
	}
	return a;
}

int main() {
	long long A, B;
	cin >> A >> B;

	long long result = A * B / Gcd(A, B);

	cout << result << "\n";

	return 0;
}